string robot_name
bool need_task_update
bool need_path_update